packages/patternfly-3/patternfly-react/src/components/DateTimePicker/TimePicker.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormControl, InputGroup, Icon, OverlayTrigger, Popover } from '../../';
+import TimeInput from './TimeComponents/TimeInput';
+import { getDateFromTime, formatTime } from './TimeHelpers';
+
+class TimePicker extends React.Component {
+  state = {
+    value: getDateFromTime(this.props.value),
+    tmpValue: formatTime(this.props.value, this.props.locale),
+    hiddenValue: this.props.hiddenValue
+  };
+
+  setSelected = time => {
+    const newTime = getDateFromTime(time);
+    this.setState({ value: newTime, tmpValue: formatTime(newTime, this.props.locale) });
+  };
+  render() {
+    const { value, hiddenValue } = this.state;
+    const { locale, placement, id } = this.props;
+    const popover = (
+      <Popover
+        id={id}
+        className="bootstrap-datetimepicker-widget date-time-picker-pf bootstrap-timepicker-widget  dropdown-menu"
+      >
+        <ul className="list-unstyled">
+          <li className="picker-switch accordion-toggle">
+            <table className="table-condensed">
+              <tbody>
+                <tr />
+              </tbody>
+            </table>
+          </li>
+          <li>
+            <TimeInput time={value} setSelected={this.setSelected} locale={locale} />
+          </li>
+        </ul>
+      </Popover>
+    );
+
+    return (
+      <InputGroup className="input-group date-time-picker-pf time-picker-pf">
+        <FormControl
+          aria-label="time-picker-input"
+          type="text"
+          value={hiddenValue ? '' : this.state.tmpValue}
```

Comment:
Yes, this component is already used in foreman.
When the picker is used inside a form, there needs to be an option for an empty value for submitting

Reply:
So then it should just be cleared right? no need to prevent re-entry or hiding the actual current value. Why have the button be a toggle? Clear if the X is pressed, disable the X if there is no entry. @mcarrano Can you take a look and make any suggestions from a UX perspective?

At a minimum this functionality should be optional with the default being to not having this functionality. I should be able to clear the text in order to unset the current time.

Reply:
I agree with @jeff-phillips-18 here. The current button that toggles between clear and edit is confusing and unexpected to me. The user should be able to just clear the field to submit a null value. In PatternFly 4, we are specifying a clear button inside the field. See https://github.com/patternfly/patternfly-next/issues/180  If you want to do something equivalent, that would be acceptable @MariaAga .

Reply:
Thanks for the feedback!
I've removed the edit/clear button and now the input field can be empty (by deleting the value). Is this a better functionality? @mcarrano